src/main/java/com/acrolinx/sidebar/utils/NamespaceResolver.java

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2023-present Acrolinx GmbH */
+package com.acrolinx.sidebar.utils;
+
+import java.util.Iterator;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+import org.w3c.dom.Document;
+
+public class NamespaceResolver implements NamespaceContext {
```

Comment:
Since this class does not add any API beyond the interface it implements, we can make the constructor private and instead add a static factory method which returns the `NamespaceContext` interface. This way we force everyone to strictly implement against the `NamespaceContext` interface (and then we should also make the class final).